pywbem/cim_obj.py

```diff
@@ -2853,6 +2853,124 @@ def tomof(self, indent=0, maxline=MAX_MOF_LINE):
 
         return u''.join(mof)
 
+    @staticmethod
+    def from_class(klass, namespace=None,
+                   property_values=None,
+                   include_null_properties=True,
+                   include_path=True, strict=False,
+                   include_class_origin=False):
+        """
+        Build a new CIMInstance from the input CIMClass using the
+        property_values dictionary to complete properties and the other
+        parameters to filter properties, validate the properties, and
+        optionally set the path component of the CIMInstance.  If any of the
+        properties in the class have default values, those values are passed
+        to the instance unless overridden by the property_values dictionary.
+        No CIMProperty qualifiers are included in the created instance and the
+        `class_origin` attribute is transfered from the class only if the
+        `include_class_origin` parameter is True
+
+        Parameters:
+          klass (:class:`pywbem:CIMClass`)
+            CIMClass from which the instance will be constructed.  This
+            class must include qualifiers and should include properties
+            from any superclasses to be sure it includes all properties
+            that are to be built into the instance. Properties may be
+            excluded from the instance by not including them in the `klass`
+            parameter.
+
+          namespace (:term:`string`):
+            Namespace in the WBEMConnection used to retrieve the class or
+            `None` if the default_namespace is to be used.
+
+          property_values (:class:`py:dict` or `NocaseDict`_):
+            Dictionary containing name/value pairs where the names are the
+            names of properties in the class and the properties are the
+            property values to be set into the instance. The values must match
+            the type defined for the property in the class. If a property is in
+            the property_values dictionary but not in the class a ValueError
+            exception is raised. Property name case independence is only
+            honored if it is a NoCaseDict (`NocaseDict`_).
```

Comment:
I don't think this requirement is acceptable when providing a case sensitive dict. How should one know what the right case is that ought to be used? The current implementation requires using the case of the properties in the klass object (and that is not even documented). But how does one know whether these are in sync with the MOF class definition?
I think this requirement should be eliminated, e.g. by rethinking the algorithm such that "in" tests are only done against NocaseDicts that already eixst, or if that cannot be achieved by copying the input dict into a NocaseDict.

Reply:
I agree.  I propose that if the input is a NoCaseDict we use it, else map the input into a NoCaseDict.

Reply:
DONE: Added code to convert any other dict type to NocaseDict and added couple more tests